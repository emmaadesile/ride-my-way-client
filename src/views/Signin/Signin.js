import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import signin from "../../actions/signin";
import rideLogo from "../../assets/img/rideLogo.png";

class Signin extends React.Component {
  state = {
    email: "",
    password: ""
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value,
      error: {}
    });
  };

  handleSubmit = event => {
    event.preventDefault();

    const { email, password } = this.state;
    const user = { email, password };
    const { loginUser } = this.props;

    loginUser(user);
  };

  render() {
    const { stateError, isAuthenticated } = this.props;

    if (isAuthenticated) {
      this.props.history.push("/");
    }
    console.log(isAuthenticated)
    return (
      <div className="wrapper2">
        <div className="form-wrapper">
          <Link to="/">
            <img src={rideLogo} alt="logo" className="ride-logo" />
          </Link>
          <form
            className="form signin-form"
            style={{ alignSelf: "start" }}
            onSubmit={this.handleSubmit}
          >
            <label className="form-label" htmlFor="signin">
              Login
            </label>
            {stateError ? (
              <div className="errorMessage">{stateError}</div>
            ) : null}
            <input
              className="form-control email"
              type="email"
              placeholder="Email"
              name="email"
              onChange={this.handleChange}
              required
            />
            <input
              className="form-control password"
              type="password"
              placeholder="Password"
              name="password"
              onChange={this.handleChange}
              required
            />
            <button className="btn btn__primary btn__inline signin-btn">
              Sign In
            </button>
            <p>
              Not a member?
              <Link to="/signup">Sign up</Link>
            </p>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { auth } = state;

  return {
    loading: auth.signin.loading,
    stateError: auth.signin.error,
    message: auth.signin.message,
    isAuthenticated: auth.isAuthenticated,
  };
};

const mapDispatchToProps = dispatch => ({
  loginUser(user) {
    return dispatch(signin(user));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Signin);
