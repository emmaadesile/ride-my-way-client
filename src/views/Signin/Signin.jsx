import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import signin from "../../actions/signin";
import Loading from '../../components/Loading/Loading.jsx';
import rideLogo from "../../assets/img/rideLogo.png";

class Signin extends React.Component {
  state = {
    email: "",
    password: ""
  };

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value,
      error: {}
    });
  };

  handleSubmit = event => {
    event.preventDefault();

    const { email, password } = this.state;
    const user = { email, password };
    const { loginUser } = this.props;

    loginUser(user);
  };

  render() {
    const { stateError, isAuthenticated, loading } = this.props;

    if (isAuthenticated) {
      this.props.history.push("/");
    }

    return (
      <div className="wrapper2">
        {loading ? <Loading /> : null}
        <div className="form-wrapper">
          <Link to="/">
            <img src={rideLogo} alt="logo" className="ride-logo" />
          </Link>
          <form
            className="form signin-form"
            style={{ alignSelf: "start" }}
            onSubmit={this.handleSubmit}
          >
            <label className="form-label" htmlFor="signin">
              Login
            </label>
            {stateError ? (
              <div className="errorMessage">{stateError}</div>
            ) : null}
            <input
              className="form-control email"
              type="email"
              placeholder="Email"
              name="email"
              onChange={this.handleChange}
              required
            />
            <input
              className="form-control password"
              type="password"
              placeholder="Password"
              name="password"
              onChange={this.handleChange}
              required
            />
            <button className="btn btn__primary btn__inline signin-btn">
              Sign In
            </button>
            <p>
              Not a member?
              <Link to="/auth/signup">Sign up</Link>
            </p>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  const { auth } = state;

  return {
    loading: auth.signin.loading,
    stateError: auth.signin.error,
    message: auth.signin.message,
    isAuthenticated: auth.isAuthenticated,
  };
};

const mapDispatchToProps = dispatch => ({
  loginUser(user) {
    return dispatch(signin(user));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Signin);
