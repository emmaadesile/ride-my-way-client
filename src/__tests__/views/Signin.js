import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import toJson from 'enzyme-to-json';
import Signin from "../../views/Signin/Signin";

const mockStore = configureMockStore();
const store = mockStore();

const mockFunction = param => jest.fn(param);

describe("Signin Page", () => {
  test("renders the signin page", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Signin loginUser={mockFunction} />
      </Provider>
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Signin />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
