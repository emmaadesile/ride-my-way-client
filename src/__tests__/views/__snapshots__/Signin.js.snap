// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Page should render correctly 1`] = `<Connect(Signin) />`;
